edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -20,10 +20,6 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
 
     public class CrdWatcher<TConfig>
```

Comment:
What do you think about completely separate watcher and logic to generate k8s primitives? So all code dependable on Kubernetes client will stay in watcher but logic to determine what should be deleted, updated or created will live in a separate class. This way each class will have only one responsibility

Reply:
I agree but I want this PR to be strictly about moving the pod/service creation stuff out of the watch.